pkgs/applications/networking/mullvad/default.nix

```diff
@@ -0,0 +1,117 @@
+{ stdenv, makeWrapper, fetchurl, dpkg
+, alsaLib, atk, cairo, cups, dbus, expat, fontconfig, freetype
+, gdk_pixbuf, glib, gnome2, nspr, nss, gtk3, gtk2, at-spi2-atk
+, gsettings-desktop-schemas, gobject-introspection, wrapGAppsHook
+, xorg, nghttp2
+, libudev0-shim, glibc, curl, openssl, autoPatchelfHook
+, udev, pulseaudio, libpulseaudio, systemd, libnotify
+}:
+
+let deps = [
+    alsaLib
+    at-spi2-atk
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    fontconfig
+    freetype
+    gdk_pixbuf
+    glib
+    gnome2.GConf
+    gnome2.pango
+    gtk2
+    gtk3
+    gsettings-desktop-schemas
+    pulseaudio
+    libpulseaudio
+    libnotify
+    xorg.libX11
+    xorg.libXScrnSaver
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libxcb
+    nspr
+    nss
+    udev
+    stdenv.cc.cc
+    systemd
+  ];
+
+in
+
+stdenv.mkDerivation rec {
+  name = "mullvad-${version}";
+  version = "2019.5";
+
+  src = fetchurl {
+    url = "https://www.mullvad.net/media/app/MullvadVPN-${version}_amd64.deb";
+    sha256 = "542a93521906cd5e97075c9f3e9088c19562b127556a3de151e25bc66b11fe0b";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+    gobject-introspection
+  ];
+
+  buildInputs = deps;
+
+  dontBuild = true;
+  dontConfigure = true;
+  dontPatchELF = true;
+  dontStrip = true;
+
+  unpackPhase = "dpkg-deb -x $src pkg";
+
+  installPhase = ''
+    runHook preInstall
+
+    libdir=$out/lib/mullvad
+    mkdir -p $out/share/mullvad $out/lib $out/bin
+    mkdir -p $libdir
+
+    mv ./pkg/usr/share/* $out/share
+    mv ./pkg/usr/bin/* $out/bin
+    mv pkg/opt/Mullvad\ VPN/* $out/share/mullvad
+    cp $out/share/mullvad/*.so $libdir
+
+    rpath="$out/share/mullvad:$libdir"
+
+    patchelf \
+       --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+       --set-rpath $rpath $out/share/mullvad/mullvad-vpn
+
+    librarypath="${stdenv.lib.makeLibraryPath deps}:$libdir"
+
+    wrapProgram $out/share/mullvad/mullvad-vpn \
+        --prefix LD_LIBRARY_PATH : "$librarypath" \
+        --prefix PATH
+
+    sed -i 's|\/opt\/Mullvad.*VPN|'$out'/bin|g' $out/share/applications/mullvad-vpn.desktop
+    sed -i 's|\/opt\/Mullvad.*VPN/resources|'$out'/bin|g' $out/share/mullvad/resources/mullvad-daemon.conf
+    sed -i 's|\/opt\/Mullvad.*VPN/resources|'$out'/bin|g' $out/share/mullvad/resources/mullvad-daemon.service
```

Comment:
Can you install this to `etc/systemd/system/mullvad-daemon.service`.
Then people can use `systemd.packages` in NixOS.